src/collective/behavior/banner/browser/viewlets.py

```diff
@@ -18,12 +18,16 @@ class BannerViewlet(ViewletBase):
 
     banner_template = ViewPageTemplateFile('banner.pt')
     slider_template = ViewPageTemplateFile('slider.pt')
+    video_template = ViewPageTemplateFile('video.pt')
 
     def index(self):
         context = aq_inner(self.context)
         if ISlider.providedBy(context):
             if context.slider_relation and len(context.slider_relation) > 1:
                 return self.slider_template()
+        banner = self.find_banner()
```

Comment:
@fulv This has the effect that ``find_banner`` is called twice. I'd prefer to it only being called one (instead of memoizing). Could you merge the video-template in the banner-template and make the condition ``'banner_url' in banner`` in there?